### PR TITLE
c8d/exporter: Use WithSkipMissing

### PR DESCRIPTION
**Pull request by vvoland**
- needs: https://github.com/moby/buildkit/pull/4558 (but not strictly required)

Save the unmodified manifest list to keep the image ID of the multi-platform images when not all platforms are present.

Note: Without https://github.com/moby/buildkit/pull/4558 vendored copying the distribution source labels into the `index.json` annotations won't work: https://github.com/containerd/containerd/pull/9554/commits/b9af453f0ccc68d9ae37202712f1a9726d9cf5de (for OCI archives exported directly from buildkit).

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: daemon/containerd/image_exporter.go
Comment (thaJeztah) [inline daemon/containerd/image_exporter.go:383]: Wondering if we want to have a utility / check around to allow returning a warning.

I think it makes sense to allow exporting "whatever is in the store" (even if partial content), but possibly we want to have;

- a warning
- an option to create a new variant of the image with only the available variants (and thus; new digest)

Not exactly sure yet what the "UX" for it would be, and if that would require adding an option to the API ("export policy" or whatever)

**Comment by thaJeztah** [comment]
Related PR in containerd;

- https://github.com/containerd/containerd/pull/9554

**Comment by thaJeztah** [comment]
containerd 1.7.12 was tagged, and I updated my existing PR that updates it (and moves some dependency);

- https://github.com/moby/moby/pull/46979

that should unblock this PR

**Review by thaJeztah — APPROVED** [review]
LGTM

but perhaps @rumpl and/or @dmcgowan can give a second review

**Review by rumpl — APPROVED** [review]

**Review by dmcgowan — APPROVED** [review]

**Comment by thaJeztah** [comment]
Let me bring this one in 👍

**Pull request merged**
